common/models/container.js

```diff
@@ -294,44 +307,56 @@ module.exports = function(Container) {
               }
             })
 
-
+            //this gets processed before the previous then block is completed
+            //TODO: promisify it properly
             .then(function () {
               excelRows = [];
               console.log("done!");
               next();
             });
         });
     }
-    else if(fileExtension[1].toLowerCase() === 'csv'){
+    else if (fileExtension[1].toLowerCase() === 'csv') {
 
+      log.debug('Converting csv to json');
       var Converter = require('csvtojson').Converter;
-      var converter = new Converter({constructResult:true}); //new converter instance
+      var converter = new Converter({constructResult: true}); //new converter instance
 
       //record_parsed will be emitted each time a row has been parsed.
-      converter.on('record_parsed', function(resultRow, rawRow, rowIndex) {
+      converter.on('record_parsed', function (resultRow, rawRow, rowIndex) {
         // map header/column names from the CSV to the field names from StockOrderLineitemModel
         resultRow['supplyPrice'] = resultRow['supplier_cost'];
         resultRow['orderQuantity'] = resultRow['quantity'];
+        //TODO: handle blank rows
+        resultRow['orderQuantity'] = resultRow['Quantity'];
+        resultRow['sku'] = resultRow['SKU'];
+        delete resultRow['SKU'];
+        delete resultRow['Quantity'];
         delete resultRow['supplier_cost'];
         delete resultRow['quantity'];
       });
 
+      converter.on('error', function (error) {
+        log.error('Could not parse csv information', error);
+        next('Could not parse csv information');
+      });
+
       //end_parsed will be emitted once parsing finished
       converter.on('end_parsed', function (arrayOfCsvRowsAsObjects) {
         log.debug('parsed csv rows:', arrayOfCsvRowsAsObjects.length);
 
         log.debug('#1 create a new Reportmodel');
-        createReportModelForCsv(item.name, Container, next)
```

Comment:
I wonder why `next` was passed explicitly here ... is removing it the right thing to do?

Reply:
I think the I have promisified the function properly. Unless there was a callback waiting to be called when this function(which is not the case), we would be in a problem.